pkg/operator/options/options.go

```diff
@@ -134,25 +135,32 @@ func (o *Options) Parse(args ...string) (*Options, error) {
 		o.FeatureGatesSet = true
 	}
 
-	return o, nil
+	ctx = ToContext(ctx, o)
+	return ctx, nil
 }
 
 // MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
 // a CLI argument or environment variable, that value will be used.
-func (o *Options) MergeSettings(s *settings.Settings) *Options {
-	if s == nil {
-		return o
+func (*Options) MergeSettings(ctx context.Context, injectables ...settings.Injectable) context.Context {
+	for _, in := range injectables {
+		_, ok := in.(*settings.Settings)
+		if !ok {
+			continue
+		}
+		s := in.FromContext(ctx).(*settings.Settings)
+		o := FromContext(ctx)
+		mergeField(&o.BatchMaxDuration, s.BatchMaxDuration, o.BatchMaxDurationSet)
+		mergeField(&o.BatchIdleDuration, s.BatchIdleDuration, o.BatchIdleDurationSet)
+		mergeField(&o.FeatureGates.Drift, s.DriftEnabled, o.FeatureGatesSet)
+		ctx = ToContext(ctx, o)
```

Comment:
You should break after you have done this in order to short-circuit

Reply:
I realized I really don't need this loop at all, rather than going over all settings injectables, it can just use the concrete settings directly...not sure why I got it stuck in my head that the abstraction was needed here.